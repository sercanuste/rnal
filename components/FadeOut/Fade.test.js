import React from "react";
import { View } from "react-native";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import FadeOut from "./index";

describe("Testing FadeOut component component", () => {
    it("renders visible", () => {
        const Wrapper = shallow(<FadeOut startWhen={true} />);
        expect(Wrapper).toMatchSnapshot();
    });

    it("renders invisible", () => {
        const Wrapper = shallow(<FadeOut startWhen={false} />);
        expect(Wrapper).toMatchSnapshot();
    });

    it("makes have children", () => {
        const wrapper = shallow(
            <FadeOut>
                <View testID="card" />
            </FadeOut>
        );
        expect(wrapper.findWhere(node => node.prop("testID") === "card")).toExist();
    });

    jest.mock("Animated", () => {
        const ActualAnimated = require.requireActual("Animated");
        return {
            ...ActualAnimated,
            timing: (value, config) => {
                return {
                    start: callback => {
                        value.setValue(config.toValue);
                        callback && callback();
                    }
                };
            }
        };
    });
});
// img, title, location, description

// import React from 'react';
// import renderer from 'react-test-renderer';

// import { AnimatedOpacityController, AnimatedOpacity } from '../AnimatedOpacity';
