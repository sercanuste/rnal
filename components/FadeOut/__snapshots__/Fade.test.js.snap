// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FadeOut component component renders invisible 1`] = `
<AnimatedComponent
  duration={500}
  style={
    Object {
      "opacity": 0,
    }
  }
/>
`;

exports[`Testing FadeOut component component renders visible 1`] = `
<AnimatedComponent
  duration={500}
  style={
    Object {
      "opacity": 0,
    }
  }
/>
`;

exports[`renders invisible 1`] = `
<AnimatedComponent
  duration={500}
  style={
    Object {
      "opacity": 0,
    }
  }
/>
`;

exports[`renders visible 1`] = `
<AnimatedComponent
  duration={500}
  style={
    Object {
      "opacity": 0,
    }
  }
/>
`;
